designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/BaseEndpointDefinitions.scala

```diff
@@ -106,36 +88,16 @@ object BaseEndpointDefinitions {
 
   private object Codecs {
 
-    implicit val authenticationErrorCodec: Codec[String, CannotAuthenticateUser.type, CodecFormat.TextPlain] = {
-      Codec.string.map(
-        Mapping.from[String, CannotAuthenticateUser.type](_ => CannotAuthenticateUser)(_ =>
-          "The supplied authentication is invalid"
-        )
-      )
-    }
-
-    implicit val authorizationErrorCodec: Codec[String, InsufficientPermission.type, CodecFormat.TextPlain] = {
-      Codec.string.map(
-        Mapping.from[String, InsufficientPermission.type](_ => InsufficientPermission)(_ =>
-          "The supplied authentication is not authorized to access this resource"
-        )
-      )
-    }
-
-    implicit val impersonationPermissionErrorCodec
-        : Codec[String, ImpersonationMissingPermissionError.type, CodecFormat.TextPlain] = {
+    implicit val authorizationErrorCodec: Codec[String, SecurityError, CodecFormat.TextPlain] = {
       Codec.string.map(
-        Mapping.from[String, ImpersonationMissingPermissionError.type](_ => ImpersonationMissingPermissionError)(_ =>
-          ImpersonationMissingPermissionError.errorMessage
-        )
+        Mapping.from[String, SecurityError](_ => InsufficientPermission)(s => s.errorMessage)
```

Comment:
`_ => InsufficientPermission` is suspicious here

Reply:
Ah, forgot about it, yeah, it is shady as hell